pkg/client/leaderelection/leaderelection.go

```diff
@@ -306,4 +311,33 @@ func BindFlags(l *componentconfig.LeaderElectionConfiguration, fs *pflag.FlagSet
 	fs.DurationVar(&l.RetryPeriod.Duration, "leader-elect-retry-period", l.RetryPeriod.Duration, ""+
 		"The duration the clients should wait between attempting acquisition and renewal "+
 		"of a leadership. This is only applicable if leader election is enabled.")
+	fs.StringVar(&l.LockType, "leader-elect-lock-type", l.LockType, fmt.Sprintf(""+
+		"Indicates the type of locking to use for leadership election. Supported options "+
+		"are `%s` (default) and `%s`.", LockTypeConfigMaps, LockTypeEndpoints))
+}
+
+func GetLock(l componentconfig.LeaderElectionConfiguration, client cs.Clientset, rlc rl.ResourceLockConfig) (rl.Interface, error) {
+	name := "kube-controller-manager"
+	switch l.LockType {
+	case LockTypeEndpoints:
+		return &rl.EndpointsLock{
+			EndpointsMeta: metav1.ObjectMeta{
+				Namespace: metav1.NamespaceSystem,
```

Comment:
This doesn't belong in this package.  This has nothing to do with the generic leader election package.